typings/entry.d.ts

```diff
@@ -18,20 +18,20 @@ export interface EntrySys extends MetaSysProps {
   publishedCounter?: number,
 }
 
-export interface Entry
-  extends EntryProp,
-    DefaultElements<EntryProp & EntrySys>,
-    MetaSys<EntrySys> {
-  archive(): Promise<Entry>,
+export interface Entry<TEntryFields = KeyValueMap>
+  extends EntryProp<TEntryFields>,
+  DefaultElements<EntryProp<TEntryFields> & EntrySys>,
+      MetaSys<EntrySys> {
+  archive<TFieldType>(): Promise<Entry<TFieldType>>,
```

Comment:
I think you can drop the generic `TFieldType` entirely here, and pass through the `TEntryFields` from the outer scope so that an entry of a certain type will always yield updated versions of the same type.
Otherwise, the type hints for the Entry do not propagate to its updated versions and you need to redefine the `FieldType` explicitly.